src/Vector3.h

```diff
@@ -393,11 +393,9 @@ class VectorHasher
 	{
 		// Guaranteed to have no hash collisions for any 128x128x128 area
 		size_t Hash = 0;
-		Hash ^= static_cast<size_t>(a_Vector.x);
-		Hash <<= 8;
-		Hash ^= static_cast<size_t>(a_Vector.y);
-		Hash <<= 8;
-		Hash ^= static_cast<size_t>(a_Vector.z);
+		Hash ^= a_Vector.x + 0x9e3779b9 + (Hash << 6) + (Hash >> 2);
+		Hash ^= a_Vector.y + 0x9e3779b9 + (Hash << 6) + (Hash >> 2);
+		Hash ^= a_Vector.z + 0x9e3779b9 + (Hash << 6) + (Hash >> 2);
```

Comment:
The collisions are 256 chunks in z axis + 1 in y, or 256 in y + 1 in x, which restricts you to 256 collisions due to max height. But on 64 bit systems, we are only reducing the entropy by 8 bits, so how about:
```
size_t Hash = 0;
Hash ^= static_cast<size_t>(a_Vector.x);
Hash <<= sizeof(size_t) * CHAR_BIT / 2;
Hash ^= static_cast<size_t>(a_Vector.z);
Hash ^= static_cast<size_t>(a_Vector.y) & 0x0F<< CHAR_BIT *sizeof(size_t) - 4;
Hash ^= static_cast<size_t>(a_Vector.y) & static_cast<size_t>(~static_cast<What>(0x0F)) << CHAR_BIT *sizeof(size_t) / 2 -4;
```
It take x and z and uses them to fill the top and bottom half of the Hash. Then we take the bottom 4 bits of y and xor them into the top 4 bits of the hash, and any bits above the bottom four with the bits starting  4 bits below halfway. Guarantees no collision within any 2048x128x2048 region, or 2048x256x2048 region if y is capped between 0 and 255 on systems with a 32 bit size_t. On systems with a 64 bit size_t it has no collisions in a 134217728x128x134217728 / 134217728x256x134217728 area. I think 134217728x128x134217728 is sufficiently far apart to prevent players from causing such an attack without teleport permisions. Based on the wiki, assuming idea conditions, it would take nearly 39 days to achieve one set of collisions.